project.py

```diff
@@ -421,17 +455,21 @@ def write_image(self, cnt, image, mask):
         """
         if cnt is None:
             return
-        rect = self.get_sorted_rectangle(cnt)
-        warp, warp_mask = self.get_cropped_image(rect, image, mask)
-
-        if(self.check_change(warp, warp_mask)):
-            cv.imwrite("slide{0}.png".format(self.number_of_slides), warp)
-            cv.imwrite("mask.png", warp_mask)
-            self.last_saved_slide = warp
-            self.last_saved_slide_mask = warp_mask
+        self.stitch_board(board)
+        if(self.check_change(board, mask)):
```

Comment:
@Foxiy I understand that you like parentheses but I would suggest you reconsider leaving a space after keywords (such as `if`).